src/changelog.ts

```diff
@@ -17,15 +17,15 @@ interface Options {
 
 export default class Changelog {
   private readonly config: Configuration;
-  private github: GithubAPI;
+  private readonly gitHosting: GitHostingAPI;
   private renderer: MarkdownRenderer;
 
   constructor(config: Configuration) {
     this.config = config;
-    this.github = new GithubAPI(this.config);
```

Comment:
Abstracted the existing `GithubAPI` to `GitHostingAPI` and added `GithubAPI` and `GitEnterpriseAPI` to implement it. If we support other services such as bitbucket and gitlab later, we can support it if we implement `GitHostingAPI` accordingly.

And since it's the responsibility of `Configuration` to decide which instance to use, I changed it to be injected through `Configuration`.